### PR TITLE
[9.101-x-prod] Update quarkus extensions to the right groupId: io.quarkus

### PR DESCRIPTION
**Pull request by wmedvede**
**Description of the change:**

**Motivation for the change:**

**Checklist**

- [X] Add or Modify a unit test for your change
- [X] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by domhanak — APPROVED** [review]

**Comment by domhanak** [comment]
Here is the PR from last time for reference https://github.com/kiegroup/kogito-serverless-operator/pull/53

File: bundle.prod/manifests/logic-operator-rhel8-controllers-config_v1_configmap.yaml [hunk 1]
```diff
@@ -32,10 +32,10 @@ data:
     # Quarkus extensions required for workflows persistence. These extensions are used by the SonataFlow build system,
     # in cases where the workflow being built has configured postgresql persistence.
     postgreSQLPersistenceExtensions:
-      - groupId: com.redhat.quarkus.platform
```
Comment (rgdoliveira) [inline bundle.prod/manifests/logic-operator-rhel8-controllers-config_v1_configmap.yaml:35]: The groupid of RHBQ was changed to com.redhat.quarkus.platform as part of https://github.com/kiegroup/kogito-serverless-operator/pull/74

File: config/manager/prod/controllers_cfg.yaml [hunk 2]
```diff
@@ -29,10 +29,10 @@ builderConfigMapName: "logic-operator-rhel8-builder-config"
 # Quarkus extensions required for workflows persistence. These extensions are used by the SonataFlow build system,
 # in cases where the workflow being built has configured postgresql persistence.
 postgreSQLPersistenceExtensions:
```
Comment (rgdoliveira) [inline config/manager/prod/controllers_cfg.yaml:31]: This one was already in midstream main like this.

**Review by rgdoliveira — COMMENTED** [review]
Shouldn't we use com.redhat.quarkus.platform instead of io.quarkus as we are using RHBQ product?

**Comment by rgdoliveira** [comment]
to answer myself, those Quarkus artifacts only exist on io.quarkus groupID, so I'm approving the PR

**Review by rgdoliveira — APPROVED** [review]

**Pull request merged**